runelite-client/src/main/java/net/runelite/client/config/ConfigManager.java

```diff
@@ -175,24 +178,52 @@ public void load()
 		{
 			log.warn("Unable to update configuration on disk", ex);
 		}
+
+		if (configWatcherTask != null)
+		{
+			configWatcherTask.cancel(true);
+		}
+
+		configWatcherTask = executor.scheduleAtFixedRate(new ConfigWatcher(propertiesFile, () ->
+		{
+			final Properties properties = new Properties();
+			loadProperties(properties);
```

Comment:
This should fail more immediately if it can't load config so it won't delete everything